core/ledger/kvledger/txmgmt/txmgr/tx_simulator.go

```diff
@@ -118,6 +118,19 @@ func (s *txSimulator) DeletePrivateData(ns, coll, key string) error {
 	return s.SetPrivateData(ns, coll, key, nil)
 }
 
+// PurgePrivateData implements method in interface `ledger.TxSimulator`
+func (s *txSimulator) PurgePrivateData(ns, coll, key string) error {
+	if err := s.queryExecutor.validateCollName(ns, coll); err != nil {
+		return err
+	}
+	if err := s.checkWritePrecondition(key, nil); err != nil {
+		return err
+	}
+	s.writePerformed = true
+	s.rwsetBuilder.AddToHashedWriteSetPurge(ns, coll, key)
+	return nil
```

Comment:
Don't you need to call checkPrivateStateMetadata(), similar to how DeletePrivateData() / SetPrivateData() does it?

Reply:
I don't think this call is necessary as the purge key is only written to the hashed read write set